File: ruby/lib/steam/community/css/css_map.rb
# This code is free software; you can redistribute it and/or modify it under
# the terms of the new BSD License.
#
# Copyright (c) 2010-2011, Sebastian Staudt

# Represents the stats for a Counter-Strike: Source map for a specific user
#
# @author Sebastian Staudt
class CSSMap

  # Returns the name of this map
  #
  # @return [String] The name of this map
  attr_reader :name

  # Returns the number of rounds the player has lost on this map
  #
  # @return [Fixnum] The number of rounds lost
  attr_reader :rounds_lost

  # Returns the number of rounds the player has played on this map
  #
  # @return [Fixnum] The number of rounds played
  attr_reader :rounds_played

  # Returns the number of rounds the player has won on this map
  #
  # @return [Fixnum] The number of rounds won
  attr_reader :rounds_won

  # Creates a new instance of a Counter-Strike: Source class based on the
  # given XML data
  #
  # @param [String] map_name The name of the map
  # @param [REXML::Element] maps_data The XML data of all maps
  def initialize(map_name, maps_data)
    @name          = map_name

    @favorite      = (maps_data.elements['favorite'].text == @name)
    @rounds_played = maps_data.elements["#{@name}_rounds"].text.to_i
    @rounds_won    = maps_data.elements["#{@name}_wins"].text.to_i

    @rounds_lost = @rounds_played - @rounds_won
    @rounds_won_percentage = (@rounds_played > 0) ? @rounds_won.to_f / @rounds_played : 0
  end

  # Returns whether this map is the favorite map of this player
  #
  # @return [Boolean] `true` if this is the favorite map
  def favorite?
    @favorite
  end

  # Returns the percentage of rounds the player has won on this map
  #
  # @return [Float] The percentage of rounds won
  def rounds_won_percentage
    (@rounds_played > 0) ? @rounds_won.to_f / @rounds_played : 0
  end

end

File: php/lib/steam/community/css/CSSMap.php
<?php
/**
 * This code is free software; you can redistribute it and/or modify it under
 * the terms of the new BSD License.
 *
 * Copyright (c) 2010, Sebastian Staudt
 *
 * @author     Sebastian Staudt
 * @license    http://www.opensource.org/licenses/bsd-license.php New BSD License
 * @package    Steam Condenser (PHP)
 * @subpackage Steam Community
 */

/**
 * CSSMap holds statistical information about maps played by a player in
 * Counter-Strike: Source.
 *
 * @package Steam Condenser (PHP)
 * @subpackage Steam Community
 */
class CSSMap {

    private $favorite;

    private $name;

    private $roundsPlayed;

    private $roundsLost;

    private $roundsWon;

    private $roundsWonPercentage;

    /**
     * Creates a new instance of CSSMap based on the assigned map name and XML
     * data
     *
     * @param $mapName
     * @param $mapsData
     */
    public function __construct($mapName, $mapsData) {
        $this->name = $mapName;

        $this->favorite     = ((string) $mapsData->favorite) == $this->name;
        $this->roundsPlayed = (int) $mapsData->{"{$this->name}_rounds"};
        $this->roundsWon    = (int) $mapsData->{"{$this->name}_wins"};

        $this->roundsLost          = $this->roundsPlayed - $this->roundsWon;
        $this->roundsWonPercentage = ($this->roundsPlayed > 0) ? ($this->roundsWon / $this->roundsPlayed) : 0;
    }

    /**
     * Returns whether this map is the favorite map of this player
     *
     * @return boolean
     */
    public function isFavorite() {
        return $this->favorite;
    }

    public function getName() {
        return $this->name;
    }

    public function getRoundsPlayed() {
        return $this->roundsPlayed;
    }

    public function getRoundsLost() {
        return $this->roundsLost;
    }

    public function getRoundsWon() {
        return $this->roundsWon;
    }

    public function getRoundsWonPercentage() {
        return $this->roundsWonPercentage;
    }
}
?>
